ui/utils.ts

```diff
@@ -0,0 +1,267 @@
+        content: content,
+        type: 'yellow',
+        typeAnimated: true,
+        buttons: {
+            close: () => {
+            }
+        }
+    });
+}
+
+// TODO: Remove this/replace with something else. In the case of a huge log hangs the entire app
+export function setup_log_watcher(callback: (alert_text: string, alert_time: number) => void) {
+    // if the log file is not found keep trying until it is
+    if (!fs.existsSync('rotkehlchen.log')) {
+        log_searcher = setInterval(function () {
+            _setup_log_watcher(callback);
+        }, 5000);
+        return;
+    }
+    _setup_log_watcher(callback);
+}
+
+export function setup_client_auditor() {
+    if (!client_auditor) {
+        client_auditor = setInterval(periodic_client_query, 60000);
+    }
+}
+
+
+export function reload_table_currency_val(table: DataTables.Api, colnum: number) {
+    table.rows().invalidate();
+    $(table.column(colnum).header()).text(settings.main_currency.ticker_symbol + ' value');
+    table.draw();
+}
+
+export function reload_table_currency_val_if_existing(table: DataTables.Api, colnum: number) {
+    if (table) {
+        reload_table_currency_val(table, colnum);
+    }
+}
+
+export function string_capitalize(s: string) {
+    return s && s[0].toUpperCase() + s.slice(1);
+}
+
+export function date_text_to_utc_ts(txt: string) {
+    // for now assuming YYY/MM/DD HH:MM
+    if (settings.datetime_format !== 'd/m/Y G:i') {
+        throw new Error('Invalid datetime format');
+    }
+    const m = txt.match(/\d+/g);
+
+    if (!m) {
+        throw new Error('match failed for ' + txt);
+    }
+
+    const day = parseInt(m[0], 10);
+    const month = parseInt(m[1], 10) - 1;
+    const year = parseInt(m[2], 10);
+    const hours = parseInt(m[3], 10);
+    const seconds = parseInt(m[4], 10);
+    return (new Date(Date.UTC(year, month, day, hours, seconds))).getTime() / 1000;
+}
+
+interface MenuItem {
+    readonly name: string;
+    readonly icon: string;
+}
+
+export function dt_edit_drawcallback(
+    id: string,
+    edit_fn?: ((row: DataTables.RowMethods) => void) | null,
+    delete_fn?: ((row: DataTables.RowMethods) => void) | null
+) {
+    return () => {
+        const ctx_menu_items: { [key: string]: string | MenuItem } = {};
+
+        if (edit_fn) {
+            ctx_menu_items['edit'] = {name: 'Edit', icon: 'fa-edit'};
+        }
+
+        if (delete_fn) {
+            ctx_menu_items['delete'] = {name: 'Delete', icon: 'fa-trash'};
+        }
+        ctx_menu_items['sep1'] = '---------';
+        ctx_menu_items['quit'] = {name: 'Quit', icon: 'fa-sign-out'};
+
+        // idea taken from:
+        // https://stackoverflow.com/questions/43161236/how-to-show-edit-and-delete-buttons-on-datatables-when-right-click-to-rows
+        $.contextMenu({
+            selector: `#${id}_body tr td`,
+            callback: (key: string) => {
+                // @ts-ignore
+                const tr = $(this).closest('tr');
```

Comment:
This was the reason for the otc trade edit not working. `this` here actually refers to the module created by the `utils.ts` file so `$(this)` doesn't make actual sense.